src/or/or.h

```diff
@@ -1745,6 +1745,27 @@ typedef struct edge_connection_t {
   uint64_t dirreq_id;
 } edge_connection_t;
 
+/**
+ * Struct to track a connection that we closed that the other end
+ * still thinks is open. Exists in origin_circuit_t::half_streams until
+ * we get an end for this stream id.
+ */
+typedef struct half_edge_t {
+  /** stream_id for the half-closed connection */
+  streamid_t stream_id;
+
+  /** How many sendme's can the other end still send, based on how
+   * much data we had sent at the time of close */
+  int sendmes_pending;
```

Comment:
It disturbs me that we're using int for a value in [0, 10], but that's out of scope for this ticket.

Reply:
Yeah, int is the type used for all of the circ and stream windows. I simply used the same type as we do there.